mitiq/cdr/cdr.py

```diff
@@ -158,7 +158,7 @@ def execute_with_cdr(
     results = executor.evaluate(to_run, observable)
     noisy_results = np.array(results).reshape(all_circuits_shape)
 
-    results = simulator.evaluate(all_circuits[0], observable)
+    results = simulator.evaluate(training_circuits, observable)
     ideal_results = np.array(results)
 
     # Do the regression.
```

Comment:
When doing the regression here it appears it is still being done on all the noisy circuits. 

This should just be done on the training circuits so it should be something like, the transpose is to ensure the shape is fitted in the linear_fit_function:

`noisy_results[1:,:].T,`

Reply:
Makes sense to me: the fit must be based only on the training (near-clifford) results. The results associated to the circuit of interest should be used only in the last line, where the final estimate is returned.

What I don't understand is why `curve_fit` doesn't give an error if `len(noisy_results)` is not equal to `len(ideal_results)`. Anyway, this is not important I guess.

Reply:
Thats right, sorry for any confusion. 

Yes I also dont know why it does not give an error... pretty weird.